pkg/feeder/feeder.go

```diff
@@ -51,19 +52,34 @@ func NewClient(baseURL, baseAPI string, client *HttpClient) *Client {
 	}
 
 	// retry mechanism for do requests
-	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient, err error) (*http.Response, error) {
+	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient) (*http.Response, error) {
 		var res *http.Response
-		for i := 0; err != nil && i < 2; i++ {
-			time.Sleep(time.Second * 10)
+		wait := 5 * time.Second
+		for i := 0; i < 10; i++ {
 			res, err = httpClient.Do(req)
+			if err != nil {
+				Logger.With("Waiting:", wait.Seconds()).Info("Waiting to do again a request")
+				time.Sleep(wait)
+				wait = wait * 2
+				continue
+			}
+			if res.StatusCode == http.StatusOK {
+				break
+			}
 		}
 		return res, err
 	}
 
-	return &Client{BaseURL: u, BaseAPI: baseAPI, httpClient: client, retryFuncForDoReq: retryFuncForDoReq}
+	requestInParallel := 1
+
+	available := make(chan bool, requestInParallel)
+	for i := 0; i < requestInParallel; i++ {
+		available <- true
+	}
+	return &Client{BaseURL: u, BaseAPI: baseAPI, httpClient: client, retryFuncForDoReq: retryFuncForDoReq, available: available}
 }
 
-func NewClientWithRetryFuncForDoReq(baseURL, baseAPI string, client *HttpClient, retryFunc func(req *http.Request, httpClient HttpClient, err error) (*http.Response, error)) *Client {
+func NewClientWithRetryFuncForDoReq(baseURL, baseAPI string, client *HttpClient, retryFunc func(req *http.Request, httpClient HttpClient) (*http.Response, error)) *Client {
```

Comment:
This function is no longer used anywhere, please remove.

Reply:
done